### ISSUE TITLE
Delay in getting updated dyno list

### ISSUE THREAD

**Issue by gregburek** [heroku/platform-api#11]
I have several detached run dynos:

```
> heroku ps 
...
=== run: one-off processes
run.1650 (1X): up 2014/05/18 18:31:30 (~ 22m ago): `bin/console`
run.5513 (PX): up 2014/05/18 18:31:11 (~ 22m ago): `bundle exec sidekiq -q tools -g worker_1d006aa1_c66a_4331_a197_23217d4ec7b3 -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1`
run.8605 (PX): up 2014/05/18 18:31:13 (~ 22m ago): `bundle exec sidekiq -q tools -g worker_dadd7d65_07ee_487d_96c4_b0ffbcb39248 -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1`
run.9531 (PX): up 2014/05/18 18:31:12 (~ 22m ago): `bundle exec sidekiq -q tools -g worker_37d3e8ac_aedd_4048_a9aa_e92bc3e09a1b -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1`
```

In console, I run: 

```
irb(main):040:0> puts PlatformAPI.connect_oauth(Config.heroku_platform_api_token).dyno.list(Config.heroku_app_name).map{|v| "%s: %s" % [v['name'],v['command']]}
web.1: bundle exec puma --config config/puma.rb config.ru
run.5513: bundle exec sidekiq -q tools -g worker_1d006aa1_c66a_4331_a197_23217d4ec7b3 -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1
run.9531: bundle exec sidekiq -q tools -g worker_37d3e8ac_aedd_4048_a9aa_e92bc3e09a1b -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1
run.8605: bundle exec sidekiq -q tools -g worker_dadd7d65_07ee_487d_96c4_b0ffbcb39248 -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1
run.1650: bin/console
=> nil
```

I then kill one detached run worker: `heroku ps:stop run.5513` which works according to `heroku ps`, but the platform-api gem continues to show stale data for hours. 

ps: 

```
> heroku ps 
...
=== run: one-off processes
run.1650 (1X): up 2014/05/18 18:31:29 (~ 29m ago): `bin/console`
run.8605 (PX): up 2014/05/18 18:31:12 (~ 30m ago): `bundle exec sidekiq -q tools -g worker_dadd7d65_07ee_487d_96c4_b0ffbcb39248 -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1`
run.9531 (PX): up 2014/05/18 18:31:11 (~ 30m ago): `bundle exec sidekiq -q tools -g worker_37d3e8ac_aedd_4048_a9aa_e92bc3e09a1b -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1`
```

console:

```
irb(main):041:0> puts PlatformAPI.connect_oauth(Config.heroku_platform_api_token).dyno.list(Config.heroku_app_name).map{|v| "%s: %s" % [v['name'],v['command']]}
web.1: bundle exec puma --config config/puma.rb config.ru
run.5513: bundle exec sidekiq -q tools -g worker_1d006aa1_c66a_4331_a197_23217d4ec7b3 -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1
run.9531: bundle exec sidekiq -q tools -g worker_37d3e8ac_aedd_4048_a9aa_e92bc3e09a1b -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1
run.8605: bundle exec sidekiq -q tools -g worker_dadd7d65_07ee_487d_96c4_b0ffbcb39248 -i ${DYNO:-1} -r ./lib/sidekiq_wrapper.rb -c 1
run.1650: bin/console
=> nil
```

Do I need to reinitialize the client or somehow force a reconnect to get up to date info?

**Comment by chriso0710** [comment]
A quick comment on this old issue and this behaviour, which I just tripped over: 
API response caching is still there in the new release 2.0.0. 
I am also disabling the cache like @jkakar suggested.

**Comment by mikehale** [comment]
@chriso0710 I just tried to reproduce this issue by following the steps outlined above by Greg, but it seems to be working fine and not caching the data for longer than it should. Would you mind posting  the steps you used to reproduce this bug?

**Comment by chriso0710** [comment]
Good point @mikehale. I just tried again today, but unfortunately was unable to reproduce it. 
It might have something to do with me updating from the deprecated legacy API (https://github.com/heroku/heroku.rb) to platform-api. I got back old dyno data after scaling up via platform-api. After adding moneta to connect_oauth the dyno list data was correct.

This works for me today:
```
@heroku = PlatformAPI.connect_oauth(ENV['HEROKU_TOKEN'])
@heroku.dyno.list(@app)
@heroku.formation.update(@app, "someprocesstype", {:quantity => 1})
@heroku.dyno.list(@app)
```

Did not run with EXCON_DEBUG though. 
Sorry for the confusion!
